src/cascadia/TerminalCore/terminalrenderdata.cpp

```diff
@@ -188,6 +206,23 @@ void Terminal::SelectNewRegion(const til::point coordStart, const til::point coo
     SetSelectionEnd(realCoordEnd, SelectionExpansion::Char);
 }
 
+void Terminal::SelectSearchRegions(std::vector<til::inclusive_rect> rects)
+{

```

Comment:
@lhecker do we need to use any of your locks here?

Reply:
I'd say no. I moved all locking calls to outside of the `Terminal` class to make the code more consistent. If you were talking about the `_assertLocked()`, then I'd also say no, because this will be indirectly covered through the `_VisibleStartIndex` (we could add `_assertLocked()` to all member functions in the future if we wanted to).